code/game/machinery/autodoc.dm

```diff
@@ -0,0 +1,149 @@
+/obj/machinery/autodoc
+	var/mob/living/carbon/occupant
+	var/datum/autodoc/autodoc_processor
+	var/locked
+	name = "Autodoc"
+	icon = 'icons/obj/autodoc.dmi'
+	icon_state = "powered_off"
+	density = 1
+	anchored = 1
+
+	circuit = /obj/item/weapon/circuitboard/autodoc
+	use_power = 1
+	idle_power_usage = 60
+	active_power_usage = 10000
+
+/obj/machinery/autodoc/New()
+	. = ..()
+	autodoc_processor = new/datum/autodoc/capitalist_autodoc()
+	autodoc_processor.holder = src
+	autodoc_processor.damage_heal_amount = 20
+
+
+/obj/machinery/autodoc/relaymove(mob/user as mob)
+	if (user.stat)
+		return
+	src.go_out()
+	return
+
+/obj/machinery/autodoc/attackby(obj/item/I, mob/living/user)
+	if(default_deconstruction(I, user))
+		return
+	if(default_part_replacement(I, user))
+		return
+	..()
+
+/obj/machinery/autodoc/verb/eject()
+	set src in view(1)
+	set category = "Object"
+	set name = "Eject Autodoc"
+
+	if (usr.incapacitated())
+		return
+	src.go_out()
+	add_fingerprint(usr)
+	return
+
+/obj/machinery/autodoc/verb/move_inside()
+	set src in view(1)
+	set category = "Object"
+	set name = "Enter Autodoc"
+
+	if(usr.stat)
+		return
+	if(src.occupant)
+		usr << SPAN_WARNING("The autodoc is already occupied!")
+		return
+	if(usr.abiotic())
+		usr << SPAN_WARNING("The subject cannot have abiotic items on.")
+		return
+	set_occupant(usr)
+	src.add_fingerprint(usr)
+	return
+
+/obj/machinery/autodoc/proc/go_out()
+	if (!occupant || locked)
+		return
+	if(autodoc_processor.active)
+		to_chat(usr, SPAN_WARNING("Autodoc is locked down! Abort all oberations if you need to go out or wait untill all operations would be done."))
+		return
+	for(var/obj/O in src)
+		O.forceMove(loc)
+	occupant.forceMove(loc)
+	occupant.reset_view()
+	occupant.unset_machine()
+	occupant = null
+	autodoc_processor.set_patient(null)
+	update_use_power(1)
+	update_icon()
+
+/obj/machinery/autodoc/proc/set_occupant(var/mob/living/L)
+	L.forceMove(src)
+	src.occupant = L
+	src.add_fingerprint(usr)
+	if(stat & (NOPOWER|BROKEN))
+		autodoc_processor.set_patient(L)
+		ui_interact(L)
+		update_use_power(2)
+		L.set_machine(src)
+	update_icon()
+
+/obj/machinery/autodoc/affect_grab(var/mob/user, var/mob/target)
+	if (src.occupant)
+		user << SPAN_NOTICE("The autodoc is already occupied!")
```

Comment:
Use `to_chat`. I'm fairly sure that with goonchat being merged, none of those << messages would even display.

Reply:
> Use `to_chat`. I'm fairly sure that with goonchat being merged, none of those << messages would even display.

Isn't my code, ok, i will replace it